### PR TITLE
Cancel sync job in middle of ufs calls if client cancelled

### PR DESCRIPTION
**Pull request by bradyoo**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/underfs/UfsStatusCache.java [hunk 1]
```diff
@@ -210,22 +213,28 @@ public UfsStatus fetchStatusIfAbsent(AlluxioURI path, MountTable mountTable)
...
-            path, e);
-        if (e instanceof InterruptedException) {
-          Thread.currentThread().interrupt();
-          throw (InterruptedException) e;
+      while (true) {
+        try {
+          return prefetchJob.get(100, TimeUnit.MILLISECONDS);
+        } catch (TimeoutException e) {
+          if (rpcContext != null) {
+            rpcContext.throwIfCancelled();
```
Comment (yuzhu) [inline core/server/master/src/main/java/alluxio/underfs/UfsStatusCache.java:226]: is there a way we can test the cancellation and how our system responds to a runtimeException being thrown in the middle of a sync?

**Review by yuzhu — COMMENTED** [review]
@bradyoo Thanks for fixing this! left some comments

**Comment by bradyoo** [comment]
@yuzhu Hmm.. I was in the middle of turning the LinkedBlockingQueue into some PriorityQueue for the Executor but I'm beginning to realize that that's not a good idea because while it moves everything I want to the front of the queue, it causes everything else to not be in submission order unless I give unique priorities to everything so I'm scrapping the implementation. I'm thinking of just having yet another Executor to handle this. What do you think?

**Comment by yuzhu** [comment]
@bradyoo i think a separate executor makes sense here.

**Comment by bradyoo** [comment]
Comments discussed and tests added

**Review by yuzhu — APPROVED** [review]

**Comment by bradyoo** [comment]
alluxio-bot, merge this please.

**Comment by bradyoo** [comment]
alluxio-bot, cherry-pick this to branch-2.4 please.

**Comment by bradyoo** [comment]
alluxio-bot, cherry-pick this to branch-2.5 please.

**Comment by bradyoo** [comment]
this is having an adverse effect in sync performance. Reverting.

**Pull request merged**